app/routes/auth.js

```diff
@@ -0,0 +1,180 @@
+const express = require('express')
+const router = express.Router()
+const crypto = require('crypto')
+const { PrismaClient } = require('@prisma/client');
+const { PrismaClientKnownRequestError } = require('@prisma/client/runtime/library');
+const prisma = new PrismaClient();
+const passport = require('passport')
+const LocalStrategy = require('passport-local')
+const passwordValidator = require('password-validator')
+const emailValidator = require('email-validator')
+
+// Configuring passport strategy.
+passport.use(new LocalStrategy(async function verify(email, password, done) {
+    let user = await prisma.user.findUnique({
+        where: {
+            email: email
+        }
+    })
+    if (!user) return done(null, false, { message: 'Incorrect login credentials.' })
+
+    crypto.pbkdf2(password, user.salt, 310000, 32, 'sha256', async (err, hashedPassword) => {
+        if (err) return done(err)
+        if (!crypto.timingSafeEqual(user.hashed_password, hashedPassword)) {
+            return done(null, false, { message: 'Incorrect login credentials.' });
+        }
+
+        return done(null, user)
+    })
+}))
+
+// Serialization Functions
+passport.serializeUser(function (user, done) {
+    process.nextTick(function () {
+        done(null, { id: user.id, email: user.email, fname: user.fname, lname: user.lname });
+    })
+});
+
+passport.deserializeUser(function (user, done) {
+    process.nextTick(function () {
+        return done(null, user);
+    })
+});
+
+// Middleware that redirects to login page if user is not logged in.
+function isLoggedIn(req, res, next) {
+    req.user ? next() : res.redirect('/login')
+}
```

Comment:
This may be useful enough/used elsewhere to warrant being its own middleware file even though it's quite simple. Just a thought.

Reply:
Moved to module middleware.js.